extensions/websocket-jsr356/deployment/src/main/java/org/apache/camel/quarkus/component/websocket/jsr356/deployment/WebSocketJSR356Processor.java

```diff
@@ -0,0 +1,80 @@
+ */
+package org.apache.camel.quarkus.component.websocket.jsr356.deployment;
+
+import java.util.List;
+
+import javax.websocket.server.ServerEndpointConfig;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.undertow.deployment.ServletContextAttributeBuildItem;
+import io.quarkus.undertow.deployment.ServletDeploymentManagerBuildItem;
+import io.undertow.websockets.jsr.DefaultContainerConfigurator;
+import io.undertow.websockets.jsr.WebSocketDeploymentInfo;
+import org.apache.camel.quarkus.component.websocket.jsr356.CamelWebSocketJSR356Config;
+import org.apache.camel.quarkus.component.websocket.jsr356.CamelWebSocketJSR356Recorder;
+import org.apache.camel.quarkus.core.deployment.CamelBeanBuildItem;
+import org.apache.camel.websocket.jsr356.JSR356WebSocketComponent;
+
+class WebSocketJSR356Processor {
+
+    private static final String FEATURE = "camel-websocket-jsr356";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    ServiceProviderBuildItem registerConfiguratorServiceProvider() {
+        // TODO: Remove this. See https://github.com/quarkusio/quarkus/issues/7509
+        return new ServiceProviderBuildItem(ServerEndpointConfig.Configurator.class.getName(),
+                DefaultContainerConfigurator.class.getName());
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    public void createWebsocketEndpoints(List<ServletContextAttributeBuildItem> servletContext,
+            CamelWebSocketJSR356Recorder recorder, CamelWebSocketJSR356Config config) {
+        ServletContextAttributeBuildItem wsDeploymentInfoAttribute = servletContext
+                .stream()
+                .filter(context -> context.getKey().equals(WebSocketDeploymentInfo.ATTRIBUTE_NAME))
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException(
+                        "Servlet context attribute: " + WebSocketDeploymentInfo.ATTRIBUTE_NAME + " not found"));
+
+        recorder.configureWebsocketEndpoints((WebSocketDeploymentInfo) wsDeploymentInfoAttribute.getValue(), config);
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    public void registerServerContainer(ServletDeploymentManagerBuildItem deploymentManager,
+            CamelWebSocketJSR356Recorder recorder) {
+        recorder.registerServerContainer(deploymentManager.getDeploymentManager());
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    public CamelBeanBuildItem createWebSocketComponent(CamelWebSocketJSR356Recorder recorder) {
+        return new CamelBeanBuildItem("websocket-jsr356", JSR356WebSocketComponent.class.getName(),
+                recorder.createJsr356Component());
+    }
```

Comment:
I think we could produce a `CamelServiceFilterBuildItem` like we do e.g. here https://github.com/apache/camel-quarkus/blob/master/extensions/bean-validator/deployment/src/main/java/org/apache/camel/quarkus/component/bean/validator/deployment/BeanValidatorProcessor.java#L38-L45

Reply:
Latest commit adds a build step for that.